### PR TITLE
DDT-971: Minor fixes

### PR DESCRIPTION
**Pull request by Abdiiir**
These fixes relate to issues found when working on [DDT-875](https://dcsa.atlassian.net/browse/DDT-875)

### CODE REVIEW

File: src/main/java/org/dcsa/core/events/edocumentation/service/impl/BookingServiceImpl.java [hunk 1]
```diff
@@ -76,6 +77,11 @@
     if (bookingID == null) return Mono.just(Collections.emptyList());
     return commodityRepository
         .findByBookingID(bookingID)
+      .switchIfEmpty(
+        Mono.error(
+          ConcreteRequestErrorMessageException.notFound(
+            "Could not find commodities for the given booking ID"
+              + bookingID)))
```
Comment (pimg) [inline src/main/java/org/dcsa/core/events/edocumentation/service/impl/BookingServiceImpl.java:84]: BookingID is a field we do not expose so this should not be in an error message

File: src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java [hunk 2]
```diff
@@ -88,34 +90,52 @@
   }
 
   Mono<Tuple2<TransportEvent, TransportEvent>> fetchTransportEventByTransportID(UUID transportId) {
-    return Mono.justOrEmpty(transportId)
+    return transportRepository
+        .findById(transportId)
+        .switchIfEmpty(
+            Mono.error(
+                ConcreteRequestErrorMessageException.notFound(
+                    "could not find a Transport with the ID: " + transportId)))
```
Comment (pimg) [inline src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java:98]: Same as for the transportId

File: src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java [hunk 3]
```diff
@@ -88,34 +90,52 @@
...
+                    transportEventRepository
+                        .findFirstByTransportCallIDAndEventTypeCodeAndEventClassifierCodeOrderByEventDateTimeDesc(
+                            x.getLoadTransportCallID(),
+                            TransportEventTypeCode.ARRI,
+                            EventClassifierCode.PLN)
+                        .switchIfEmpty(
+                            Mono.error(
+                                ConcreteRequestErrorMessageException.notFound(
+                                    "could not find a TransportCall with Load Transport call ID: "
+                                        + x.getLoadTransportCallID()))),
```
Comment (pimg) [inline src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java:111]: same as above

File: src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java [hunk 4]
```diff
@@ -88,34 +90,52 @@
...
+                    transportEventRepository
+                        .findFirstByTransportCallIDAndEventTypeCodeAndEventClassifierCodeOrderByEventDateTimeDesc(
+                            x.getDischargeTransportCallID(),
+                            TransportEventTypeCode.DEPA,
+                            EventClassifierCode.PLN)
+                        .switchIfEmpty(
+                            Mono.error(
+                                ConcreteRequestErrorMessageException.notFound(
+                                    "could not find a TransportCall with Discharge Transport call ID: "
+                                        + x.getDischargeTransportCallID())))));
```
Comment (pimg) [inline src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java:121]: same as above

File: src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java [hunk 5]
```diff
@@ -88,34 +90,52 @@
...
+    return fetchTransportCallByID(id)
+        .flatMap(
+            transportCall ->
+                locationService
+                    .fetchLocationByID(transportCall.getLocationID())
+                    .switchIfEmpty(
+                        Mono.error(
+                            ConcreteRequestErrorMessageException.notFound(
+                                "Could not find a Location with the given location ID"
+                                    + transportCall.getLocationID()))));
```
Comment (pimg) [inline src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java:138]: same as above

**Review by pimg — DISMISSED** [review]
some minor changes required

File: src/main/java/org/dcsa/core/events/edocumentation/service/impl/BookingServiceImpl.java (hunk 1 above)
Comment (Abdiiir) [reply src/main/java/org/dcsa/core/events/edocumentation/service/impl/BookingServiceImpl.java:84]: We should consider returning a server error as these are unexpected states.

File: src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java [hunk 6]
```diff
@@ -88,34 +89,49 @@
   }
 
   Mono<Tuple2<TransportEvent, TransportEvent>> fetchTransportEventByTransportID(UUID transportId) {
-    return Mono.justOrEmpty(transportId)
+    return transportRepository
+        .findById(transportId)
+        .switchIfEmpty(
+            Mono.error(
+                ConcreteRequestErrorMessageException.notFound(
+                    "C not find a Transport event")))
```
Comment (HenrikHL) [inline src/main/java/org/dcsa/core/events/edocumentation/service/impl/TransportServiceImpl.java:97]:
```suggestion
                    "Can not find a Transport event")))
```

**Review by mp-assco — APPROVED** [review]

**Pull request merged**